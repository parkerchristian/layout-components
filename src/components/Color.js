import React from 'react';

const hexFromColor = color => {
  return `${color.toString(16).padStart(2, 0)}`.toUpperCase();
};

const hexFromRGB = (rgb) => {
  return `#${hexFromColor(rgb.red)}${hexFromColor(rgb.green)}${hexFromColor(rgb.blue)}`;
};

export default function Color() {
  const name = 'Black';
  const rgb = {
    red: 0,
    green: 0,
    blue: 0
  };
  const hex = hexFromRGB(rgb);

  const colorStyle = {
    display: 'inline-block',
    backgroundColor: hex,
    width: '0.8em',
    height: '0.8em'
  };

  return (
    <dl>
      <dt>Name</dt>
      <dd>{name} <div style={colorStyle}></div></dd>

      <dt>Hex</dt>
      <dd>{hex}</dd>

      <dt>RGB</dt>
      <dd>
        <p>Red - {rgb.red}</p>
        <p>Green - {rgb.green}</p>
        <p>Blue - {rgb.blue}</p>
      </dd>
    </dl>
  );
}
