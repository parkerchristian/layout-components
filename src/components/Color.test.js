import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color component', () => {
  it('renders a Color', () => {
    const wrapper = shallow(<Color />);
    expect(wrapper).toMatchSnapshot();
  });
});
