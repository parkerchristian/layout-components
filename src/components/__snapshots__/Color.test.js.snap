// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color component renders a Color 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd>
    Black
     
    <div
      style={
        Object {
          "backgroundColor": "#000000",
          "display": "inline-block",
          "height": "0.8em",
          "width": "0.8em",
        }
      }
    />
  </dd>
  <dt>
    Hex
  </dt>
  <dd>
    #000000
  </dd>
  <dt>
    RGB
  </dt>
  <dd>
    <p>
      Red - 
      0
    </p>
    <p>
      Green - 
      0
    </p>
    <p>
      Blue - 
      0
    </p>
  </dd>
</dl>
`;
